publication-commons/src/main/java/no/unit/nva/publication/service/impl/PublishingService.java

```diff
@@ -0,0 +1,169 @@
+            return;
+        }
+
+        if (!userInstance.getTopLevelOrgCristinId().equals(channelClaim.get().claimedBy().organizationId())) {
+            throw new ForbiddenException();
+        }
+
+        resource.publish(resourceService, userInstance);
+    }
+
+    private Resource getResource(SortableIdentifier resourceIdentifier) throws NotFoundException {
+        return Resource.resourceQueryObject(resourceIdentifier)
+                   .fetch(resourceService)
+                   .orElseThrow(() -> new NotFoundException("Resource not found!"));
+    }
+
+    private void publishResourceWithPendingFiles(UserInstance userInstance, Resource resource)
+        throws ApiGatewayException {
+        var workflow = getCustomerWorkflow(userInstance);
+        if (resource.isDegree()) {
+            handleDegreeResource(userInstance, resource, workflow);
+        } else {
+            PublishingRequestCase.create(resource, userInstance, workflow).persistNewTicket(ticketService);
+        }
+    }
+
+    private void handleDegreeResource(UserInstance userInstance, Resource resource, PublishingWorkflow workflow)
+        throws ApiGatewayException {
+        var publisher = getPublisher(resource);
+        if (publisher.isPresent()) {
+            var channelClaim = getChannelClaim(publisher.get());
+            if (channelClaim.isPresent() && isClaimedByDifferentOrganization(channelClaim.get(), userInstance)) {
+                var organizationId = getOrganizationId(channelClaim.get());
+                FilesApprovalThesis.create(resource, userInstance, organizationId, workflow)
+                    .persistNewTicket(ticketService);
+                return;
+            }
+        }
+        FilesApprovalThesis.createForUserInstitution(resource, userInstance, workflow)
+            .persistNewTicket(ticketService);
+    }
+
+    private static URI getOrganizationId(ChannelClaimDto channelClaim) {
+        return channelClaim.claimedBy().organizationId();
+    }
+
+    private Optional<ChannelClaimDto> getChannelClaim(Publisher publisher) throws BadGatewayException {
+        try {
+            return Optional.of(identityServiceClient.getChannelClaim(createChannelClaimUri(publisher)));
+        } catch (NotFoundException exception) {
+            return Optional.empty();
+        } catch (Exception e) {
+            throw new BadGatewayException("Could not fetch channel owner!");
+        }
+    }
+
+    private boolean isClaimedByDifferentOrganization(ChannelClaimDto channelClaim, UserInstance userInstance) {
+        return !channelClaim.claimedBy().id().equals(userInstance.getCustomerId());
+    }
+
+    private URI createChannelClaimUri(Publisher publisher) {
+        var channelClaimIdentifier = UriWrapper.fromUri(publisher.getId())
+                                         .replacePathElementByIndexFromEnd(0, StringUtils.EMPTY_STRING)
+                                         .getLastPathElement();
```

Comment:
This is because channel claims currently have years?

Reply:
I'd maybe therefore comprehend the structure and optionally remove the unmatching part (to be future-proofed for the same without year or trailing slash)

Reply:
Agree, but here we are consuming publisher id from publication. Will this value also be missing `/{year}` part?